File: split/scanfilename.pas
procedure scanfilename;
label
  30;
begin
  nameinprogress := true;
  beginname;
  {406:}
repeat
    getxtoken;
  until curcmd <> 10{:406}
;
  while true do
  begin
    if (curcmd > 12) or (curchr > 255) then
    begin
      backinput;
      goto 30;
    end;
    if not morename(curchr) then
      goto 30;
    getxtoken;
  end;
  30:
    endname;
  nameinprogress := false;
end;

File: cpp/scanfilename.cpp
#include "scanfilename.h"
#include "beginname.h"
#include "getxtoken.h"
#include "backinput.h"
#include "morename.h"
#include "endname.h"

void scanfilename(void)
{
	nameinprogress = true;
	beginname();
	do
		getxtoken();
	while (curcmd == spacer);
	while (true)
	{
		if (curcmd > other_char || curchr > 255)
		{
			backinput();
			break;
		}
		if (!morename(curchr))
			break;
		getxtoken();
	}
	endname();
	nameinprogress = false;
}
